pkg/workload/tpcc/tpcc.go

```diff
@@ -1096,7 +1098,21 @@ func (w *tpcc) executeTx(
 	}
 
 	if w.txnRetries {
-		return onTxnStartDuration, crdbpgx.ExecuteTx(ctx, conn, txOpts, txnFuncWithStartFuncs)
+		for {
+			tx, err := conn.BeginTx(ctx, txOpts)
+			if err != nil {
+				return onTxnStartDuration, err
+			}
+			err = txnFuncWithStartFuncs(tx)
+			if err == nil {
+				return onTxnStartDuration, tx.Commit(ctx)
+			}
+			var pgErr *pgconn.PgError
+			if errors.As(err, &pgErr) && pgcode.MakeCode(pgErr.Code) == pgcode.SerializationFailure {
+				continue
```

Comment:
Previous mechanism had an upper bound on the number of retries. (I believe the default is `50`.) Shouldn't we enforce it here?

Reply:
Yes, good point. I'll add the maxRetries back in.